src/containers/BookForm.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { addBook } from '../actions';
+
+const categories = [
+  'Action',
+  'Biography',
+  'History',
+  'Horror',
+  'Kids',
+  'Learning',
+  'Sci-Fi',
+];
+
+class BookForm extends React.Component {
+  state = {
+    title: '',
+    category: '',
+  }
+
+  submitBook = (event) => {
+    event.preventDefault();
+    const book = this.state;
+    this.props.addBook(book);
+    this.setState({
+      title: '',
+    })
+  }
+
+  handleChange = (event) => {
+    switch (event.target.tagName) {
+      case 'INPUT':
+      this.setState({ title: event.target.value });
+      break;
+      case 'SELECT':
+      this.setState({ category: event.target.value });
+      break;
+      default:
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <input value={this.state.title} onChange={this.handleChange} />
+        <select onChange={this.handleChange}>
+          <option value='' disabled selected>Choose category</option>
```

Comment:
Including `selected` here is causing one of the errors shown in the main comment. If you want "Choose Category" to be the default selected category then this is the way to do it:

```
<select onChange={this.handleChange} value={this.state.category}>
```

And then we can remove the `selected` keyword in the first `option`.